### PR TITLE
feat(SCT-1813): Allocation link in header

### PR DESCRIPTION
**Pull request by mapleleafjack**
This PR needs to be merged AFTER https://github.com/LBHackney-IT/lbh-social-care-frontend/pull/873, as some of the new links points to these pages

**What**  
This PR adds a component that displays links to the team and allocation pages.

Screenshot 2022-04-12 at 16 25 10
Screenshot 2022-04-12 at 16 24 49
Screenshot 2022-04-12 at 16 24 41

**Why**  
To complete the UI link/loop for the allocations

**Anything else?**

- Have you added any new third-party libraries?
- Are any new environment variables or configuration values needed?
- Anything else in this PR that isn't covered by the what/why?

### CODE REVIEW

File: components/ResidentPage/AllocationHeader.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            allocatedWorker: 'Jon Doe',
+            allocatedWorkerTeam: undefined,
+          }),
+          allocationFactory.build({ allocatedWorker: 'Foo Bar' }),
+          allocationFactory.build({ allocatedWorker: 'Agent Smith' }),
+        ]}
+      />
+    );
+    expect(screen.getByTestId('allocationHeader')).toHaveTextContent(
+      'Allocated to Jon Doe and 2 other'
```
Comment (acodeninja) [inline components/ResidentPage/AllocationHeader.spec.tsx:73]: Just a tiny language thing, should be 2 others? Looking at the code in the component it should be rendering it as others anyway?

File: components/ResidentPage/AllocationHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    </Link>
+  );
+
+  const moreLink = allocations?.length > 1 && (
+    <Link href={`/residents/${allocations[0].personId}/allocations`}>
+      <a
+        data-testid="allocationLink"
+        className={`lbh-link lbh-link--no-visited-state`}
+      >
+        {allocations?.length === 2
```
Comment (acodeninja) [inline components/ResidentPage/AllocationHeader.tsx:28]: Should this be `allocations?.length =< 2`?

**Review by acodeninja — CHANGES_REQUESTED** [review]
Small change around pluralisation to do.

File: components/ResidentPage/AllocationHeader.tsx (hunk 2 above)
Comment (mapleleafjack) [reply components/ResidentPage/AllocationHeader.tsx:28]: It should be correct, as if the allocations are 2 we display the first one with long description + "one other"

If it's more than 2 (the "else") it says the length -1 (so, not counting the one with long description)

File: components/ResidentPage/AllocationHeader.spec.tsx (hunk 1 above)
Comment (mapleleafjack) [reply components/ResidentPage/AllocationHeader.spec.tsx:73]: changed from other -> others

**Review by acodeninja — APPROVED** [review]
Sweet! Thanks for sorting that.

**Pull request merged**